myhpom/models/user_details.py

```diff
@@ -14,8 +17,73 @@ class UserDetails(models.Model):
     """
 
     user = models.OneToOneField(User)
-    state = models.ForeignKey(State, on_delete=models.SET_NULL, null=True, blank=True)
-    middle_name = models.CharField(max_length=30, null=True, blank=True)
-    accept_tos = models.NullBooleanField()
-    health_network = models.ForeignKey(HealthNetwork, on_delete=models.SET_NULL, null=True)
-    custom_provider = models.CharField(max_length=1024, null=True, blank=True)
+    accept_tos = models.NullBooleanField(
+        help_text="Whether the user has accepted the Terms of Service."
+    )
+    state = models.ForeignKey(
+        State,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        help_text="The state in which the user receives health care.",
+    )
+    middle_name = models.CharField(
+        max_length=30, null=True, blank=True, help_text="The user's middle name, if any."
+    )
+    health_network = models.ForeignKey(
+        HealthNetwork,
+        on_delete=models.SET_NULL,
+        null=True,
+        help_text="The health network that the user belongs to.",
+    )
+    custom_provider = models.CharField(
+        max_length=1024,
+        null=True,
+        blank=True,
+        help_text="The name of the user's health care provider, if not in our database.",
+    )
+    health_network_updated = models.DateTimeField(
+        auto_now_add=True,
+        null=True,
+        help_text="When the user's health network selection was last updated.",
+    )
+    birthdate = models.DateField(null=True, blank=True, help_text="The user's date of birth.")
+    # gender according to the GENDER_CHOICES above
+    gender = models.CharField(
+        max_length=32,
+        null=True,
+        blank=True,
+        choices=GENDER_CHOICES,
+        help_text="The user's gender self-identification.",
+    )
+    # zip + 4 = 10 characters
+    zip_code = models.CharField(
+        max_length=10,
+        null=True,
+        blank=True,
+        help_text="The zip code for the user's health care address.",
+    )
+    # free form phone number, long enough to include ext. etc.
+    phone = models.CharField(
```

Comment:
Per our Slack discussion, it'd be nice to have some simple validation on this (form validation at least).

Reply:
ok, I added the form validations, but not any javascript / html pattern limitations.